### ISSUE TITLE
TypeError: Cannot set property 'example' of undefined

### ISSUE THREAD

**Issue by x8core** [sindresorhus/on-change#32]
Hi. OS: Windows 10, Node: v10.15.3

Execute this.
```javascript
const onChange = require('on-change');

let object = {
    example: 1,
};

const proxy = onChange(object, (path, value, previousValue) => {});

proxy.exec = proxyInFunc => proxyInFunc.example = 2;
proxy.exec(proxy);
```

And got this.
```
C:\Users\Storm\Dropbox\minipark\node_modules\on-change\index.js:76
                        item[property] = previous;
                                       ^

TypeError: Cannot set property 'example' of undefined
    at handleChange (C:\Users\Storm\Dropbox\minipark\node_modules\on-change\index.js:76:19)
    at Object.set (C:\Users\Storm\Dropbox\minipark\node_modules\on-change\index.js:156:5)
    at Proxy.proxyInEval (eval at <anonymous> (C:\Users\Storm\Dropbox\minipark\test.js:9:14), <anonymous>:1:36)
    at Object.apply (C:\Users\Storm\Dropbox\minipark\node_modules\on-change\index.js:201:28)
    at Object.<anonymous> (C:\Users\Storm\Dropbox\minipark\test.js:10:7)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
```

**Comment by DarrenPaulWright** [comment]
I found the problem. I'll get a fix in soon.

**Comment by DarrenPaulWright** [comment]
Fixed by 2c8d1e0e4fceca3c4c3ad0add0b01650d2f2c6fa
